src/androidTest/java/org/medicmobile/webapp/mobile/LoginTests.java

```diff
@@ -0,0 +1,116 @@
+import static androidx.test.espresso.web.sugar.Web.onWebView;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.clearElement;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.findElement;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.getText;
+import static androidx.test.espresso.web.webdriver.DriverAtoms.webClick;
+import static androidx.test.platform.app.InstrumentationRegistry.getInstrumentation;
+import static org.hamcrest.Matchers.containsString;
+
+@LargeTest
+@RunWith(AndroidJUnit4.class)
+@FixMethodOrder(MethodSorters.NAME_ASCENDING)
+public class LoginTests {
+
+
+	//TODO: user/pass to be encrypted
+	public static final String USERNAME = BuildConfig.TEST_USERNAME;
+	public static final String PASSWORD = BuildConfig.TEST_PASSWORD;
+	public static final String SERVER_URL = BuildConfig.SERVER_URL;
+
+	@Rule
+	public ActivityTestRule<StartupActivity> mActivityRule = new ActivityTestRule<>(
+			StartupActivity.class);
+	SharedPreferences.Editor preferencesEditor;
+
+	@Before
+	public void setUp() {
+		Context targetContext = getInstrumentation().getTargetContext();
+		preferencesEditor = PreferenceManager.getDefaultSharedPreferences(targetContext).edit();
+		preferencesEditor.putString("api-url", SERVER_URL);
+		preferencesEditor.commit();
+	}
+
+	@Test
+	public void checkLoginPageObjects() throws Exception {
+		mActivityRule.launchActivity(new Intent());
+		Thread.sleep(5000);//TODO: use better ways to handle delays
+		onWebView()
+				.check(webContent(hasElementWithId("form")))
+				.withElement(findElement(Locator.ID, "locale"))
+				// Verify that the response page contains the entered text
+				.check(webMatches(getText(), containsString("English")));
+		String[] codes = {"es", "en", "fr", "sw"};
+		for (String code : codes) {
+
+			onWebView().withElement(findElement(Locator.NAME, code))
+					.check(webMatches(getText(), containsString(GetLanguage(code))));
+		}
+	}
+
+	@Test
+	public void loginAsRestrictedUser() throws Exception {
+
+		onWebView().withElement(findElement(Locator.ID, "user"))
+				.perform(clearElement())
+				.perform(DriverAtoms.webKeys(USERNAME))	//to be created first
+				.withElement(findElement(Locator.ID, "password"))
+				.perform(clearElement())
+				.perform(DriverAtoms.webKeys(PASSWORD))
+				.withElement(findElement(Locator.ID, "login"))
+				.perform(webClick());
+		Thread.sleep(20000);//TODO: use better ways, takes some time to load depending on docs and whether it is first replication
+		onWebView()
+				.check(webContent(hasElementWithId("message-list")))
+				.check(webContent(hasElementWithId("messages-tab")))
+				.check(webContent(hasElementWithId("tasks-tab")))
+				.check(webContent(hasElementWithId("contacts-tab")))
+				.check(webContent(hasElementWithId("analytics-tab")))
+				.check(webContent(hasElementWithId("reports-tab")));
+	}
+
+	private String GetLanguage(String code) {
```

Comment:
Rename the method to `getLanguage` (first letter in a function name always is in lower case according with the Java conventions).